### DISCUSSION TITLE
Question: Upgrading from 8 to 9

### DISCUSSION THREAD

**Discussion by eric-wilson** [PiranhaCMS/piranha.core#1561] [Q&A]
Is there an upgrade doc or any known issues when upgrading from 8 to 9? Is an upgrade possible or do I need to start from scratch etc?  I haven't tried it yet.  I just want to find out what I'm getting into first.  Thx!

BTW - great work on the code base.  I finally have my site up and running.  It's still in it's early stages but it's live: https://geekcafe.com/

There is a weird issue with nginx right now, so if you see some static files (images and css) not loading - it's not the code base but something in nginx that's the problem.  Everything from the CDN works great but anything in the local /assets gives a 404 on every other request... it's super weird.

I'm going to do a write up and possibly videos on how to make it ready for AWS with load balancers and autoscaling and point out some of the gotcha's along the way 😉

**Comment by jensbrak** [comment]
I'm running a 9-based site on nginx and have successfully updated from 8.
Nothing particular happened but I do have some weird issues with client side errors due to too large cookie sizes. Haven't really digged into it since it's only client side and in manager. Nginx and Piranha/dotnet seems to have its share of... things to consider, but it does work rather well. :)

Nice site btw. I have a ... 'geek blog' running Piranha. B) (lemme know if you wanna try out my anti spam module for piranha!)

**Comment by eric-wilson** [comment]
@jensbrak , yeah let me know how I can try your anti-spam module.  I added some basic validation but I'd like to add something better.

When you updated from 8 to 9, did you need to do anything special to use the new reusable content blocks in 9? I don't see them in the manager.  I see then in the links section for content, but I don't see a way to actually generate a reusable block.

**Comment by jensbrak** [comment]
> yeah let me know how I can try your anti-spam module. I added some basic validation but I'd like to add something better.

I simply called it `SpamDetector` and it's available both here at Github and on NuGet. Either follow my profile pic or check out the official Module repo of Piranha. Not to be considered production ready yet but I use it on my own site (wrote the module more as an excercise on how to write a module than to solve an actual problem, but I picked a useful usecase for it I hope). 

The README.md should provide you what you need to get going. Still, if you have any questions or need help etc, I'm on the Piranha Gitter chat every now and then =)

> When you updated from 8 to 9, did you need to do anything special to use the new reusable content blocks in 9?

I have not yet played around with this feature but have you checked out the template projects and what they show?
